sdk/keyvault/Microsoft.Azure.KeyVault/tests/TestFramework/KeyVaultTestFixture.cs

```diff
@@ -31,7 +31,24 @@ public class KeyVaultTestFixture : IDisposable
         public string keyVersion;
         public KeyIdentifier keyIdentifier;
         public ClientCredential clientCredential;
-        public RetryPolicy retryExecutor;
+        public RetryPolicy retryExecutor
+        {
+            get
+            {
+                switch (HttpMockServer.Mode)
+                {
+                    case HttpRecorderMode.None:
+                        return new RetryPolicy<SoftDeleteErrorDetectionStrategy>(new FixedIntervalRetryStrategy(5, TimeSpan.FromSeconds(5.0)));
+                    case HttpRecorderMode.Record:
+                        return new RetryPolicy<SoftDeleteErrorDetectionStrategy>(new ExponentialBackoffRetryStrategy(8, TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(60), TimeSpan.FromSeconds(5)));
+                    case HttpRecorderMode.Playback:
+                        return new RetryPolicy<SoftDeleteErrorDetectionStrategy>(new FixedIntervalRetryStrategy(5, TimeSpan.Zero));
+                    default:
+                        throw new ArgumentOutOfRangeException();
```

Comment:
Nit: if this were ever reached, it would be easier to diagnose something like `throw new Not supported exception($"{HttpServerMock.Mode} is not supported")`.